llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -625,14 +639,31 @@ void AArch64AsmPrinter::emitHwasanMemaccessSymbols(Module &M) {
                                      .addImm(4)
                                      .addImm(55),
                                  *STI);
-    OutStreamer->emitInstruction(
-        MCInstBuilder(AArch64::LDRBBroX)
-            .addReg(AArch64::W16)
-            .addReg(IsShort ? AArch64::X20 : AArch64::X9)
-            .addReg(AArch64::X16)
-            .addImm(0)
-            .addImm(0),
-        *STI);
+
+    if (IsFixedShadow) {
+      OutStreamer->emitInstruction(MCInstBuilder(AArch64::MOVZXi)
```

Comment:
optional: leave a comment about this instruction sequence. Particularly why the `FixedShadowOffset >> 32` (and does that mean FixedShadowOffset needs to be bigger than `1 << 32`?

Reply:
Done